packages/ui/components/core/src/SlotMixin.js

```diff
@@ -131,7 +131,11 @@ const SlotMixinImplementation = superclass =>
       if (isFirstRender) {
         // @ts-expect-error wait for browser support
         const supportsScopedRegistry = !!ShadowRoot.prototype.createElement;
-        const registryRoot = supportsScopedRegistry ? this.shadowRoot || document : document;
+        const hasShadowRoot = Boolean(this.shadowRoot);
+        if (!hasShadowRoot) {
+          throw new Error(`No shadowRoot was found`);
```

Comment:
```suggestion
          // TODO: throw an error in a breaking release
          console.error(`[SlotMixin] No shadowRoot was found`);
```

Hey, thanks for you contribution :) 
Since I think there are consumers that use this mixin, throwing an error would be breaking. So I think we can best leave that for a breaking release

Reply:
Could you also add a changeset? (via `npm run hcangeset`)